src/Core/Checkout/Customer/SalesChannel/AddWishlistProductRoute.php

```diff
@@ -93,7 +97,12 @@ private function validateProduct(string $productId, SalesChannelContext $context
         $productsIds = $this->productRepository->searchIds(new Criteria([$productId]), $context);
 
         if ($productsIds->firstId() === null) {
-            throw new ProductNotFoundException($productId);
+            // @deprecated tag:v6.8.0 - remove this if block
+            if (!Feature::isActive('v6.8.0.0')) {
```

Comment:
why the feature flag here? it would only make sense, if you throw a different exception from `CustomerException::productNotFound` which you currently do not do :slightly_smiling_face: but see my other comment on that method